dotcom-rendering/src/components/Lightbox.tsx

```diff
@@ -0,0 +1,63 @@
+import type { ArticleFormat } from '../lib/articleFormat';
+import type { ImageForAppsLightbox } from '../model/appsLightboxImages';
+import type { Switches } from '../types/config';
+import type { ImageForLightbox } from '../types/content';
+import type { RenderingTarget } from '../types/renderingTarget';
+import { AppsLightboxImageStore } from './AppsLightboxImageStore.importable';
+import { Island } from './Island';
+import { LightboxHash } from './LightboxHash.importable';
+import { LightboxLayout } from './LightboxLayout.importable';
+
+interface BaseProps {
+	format: ArticleFormat;
+	renderingTarget: RenderingTarget;
+	switches: Switches;
+}
+
+interface WebProps extends BaseProps {
+	renderingTarget: 'Web';
+	lightboxImages: ImageForLightbox[];
+}
+
+interface AppProps extends BaseProps {
+	renderingTarget: 'Apps';
+	lightboxImages: ImageForAppsLightbox[];
+}
+
+export const Lightbox = ({
+	format,
+	lightboxImages,
+	renderingTarget,
+	switches,
+}: WebProps | AppProps) => {
+	switch (renderingTarget) {
+		case 'Web':
+			if (
+				switches.lightbox === undefined ||
```

Comment:
Is this check needed?

Reply:
The `lightbox` switch has three possible values: `undefined`, `true` or `false`. The first condition handles `undefined`. If it's `false` then type is narrowed to `boolean`, which the second condition handles.

Technically you could rely on JavaScript's implicit coercion behaviour to combine this all into a single condition, which would convert `undefined` into a `boolean`, but I prefer to handle these checks separately as I think it's clearer.